New view only map page

cesswairimu opened the pull request:
Fixes #761 

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/mapknitter-reviewers` for help, in a comment below

jywarren commented:
Exciting!!!

jywarren commented:
I was thinking, when we finish this, maybe we should apply it to the tag page too, /tag/featured for example! With a map of the maps with that tag.

cesswairimu commented:
@jywarren noted..Will work to get this done today

cesswairimu commented:
Hey @jywarren , what do u think?  Also where do we want the `edit this map` button to link to/do?

![screencapture-localhost-3000-m-indiana-2019-07-06-01_38_26](https://user-images.githubusercontent.com/17081074/60747608-f2408d80-9f8e-11e9-9fb0-b1f94f758d67.png)

jywarren commented:
Looking great! I think:

1. "env data near here" will have to wait until LEL's menu is done - see https://github.com/publiclab/leaflet-environmental-layers/milestone/2 and https://github.com/publiclab/leaflet-environmental-layers/issues/232
2. can we add "Nearby mappers" or should we really show "nearby maps" and just highlight who the mappers are?
3. "Edit this map" will go to the current page like https://mapknitter.org/maps/pvdtest/ where you can actually edit the map! make sense?

jywarren commented:
What if we had a section like "Share + Embed" or something?

cesswairimu commented:
Great thanks Jeff

cesswairimu commented:
I think displaying nearby maps and highlighting the mappers is seems more convenient...I will change to that.

jywarren commented on app/controllers/front_ui_controller.rb:
```diff
@@ -38,6 +38,14 @@ def save_location
     render nothing: true
   end
 
+  def view_map
```
Oh, maybe this belongs in the `maps_controller`, next to the `show` method? Thanks, Cess!

cesswairimu commented:
![screencapture-mapknitter-unstable-laboratoriopublico-org-m-central-park-nyc-2019-07-10-01_02_32](https://user-images.githubusercontent.com/17081074/60967258-e8b67d00-a322-11e9-8e4d-303835c955c9.png)

cesswairimu commented:
Hey @jywarren  could you please point me to the partial/script that renders the images on top of the map? Thanks

jywarren commented:
Ah, it should be the one we use in the `/embed/_mapname_` route. And i
believe we mark the images as "editable:false" or something like that in
JavaScript!

Aha - no, it's the same template but we set `readOnly` here:
https://github.com/publiclab/mapknitter/blob/1c7c1371fc064e64905dc4c96ebf554455a4e499/app/views/layouts/knitter2.html.erb#L197

On Wed, Jul 10, 2019 at 10:28 AM Cess <notifications@github.com> wrote:

> Hey @jywarren <https://github.com/jywarren> could you please point me to
> the partial/script that renders the images on top of the map? Thanks
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/mapknitter/pull/777?email_source=notifications&email_token=AAAF6J5ZILQPVXLHEK4KJBDP6XWXVA5CNFSM4H4WTQUKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODZTUORA#issuecomment-510084932>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAF6JYR2UTVV5JUPB7QHR3P6XWXVANCNFSM4H4WTQUA>
> .
>

cesswairimu commented:
Great thanks

cesswairimu commented:
Hey @jywarren, I removed the 'Env data near here' section until LEL is ready. I also moved the logic to maps_controller. I added an `embed code` button..let me know if this is what you had in mind? 
![screencapture-localhost-3000-m-nairobi-2019-07-11-13_40_13](https://user-images.githubusercontent.com/17081074/61044680-d2232b00-a3e1-11e9-9468-159cb9d5cbdc.png)

Also thinking for a follow-up issue maybe we can change the map design in `Map.js` to the same map design as we have for the new front-page or a another design what do you think?

jywarren commented:
Approved codeclimate! Looking for a review, Cess? 😄

cesswairimu commented:
Yes please @jywarren

cesswairimu commented:
@jywarren this is ready too..kindly review. Thanks

gauravano commented:
Hey @cesswairimu, the design looks great. Please remove merge conflicts. Will pull this PR and test it locally today.

gauravano commented:
Also, a typo in file name https://github.com/publiclab/mapknitter/pull/777/files#diff-1eec38a49b2119d29bf2d626686504c5 :balloon:

cesswairimu commented:
Thanks @gauravano..Just fixed the conflicts...can't see the typo though could you please  tell me which file it is? Thanks

gauravano commented:
file with name baloon  -- > balloon :balloon:. Sorry, :rofl:, it was a bad way to point.

cesswairimu commented:
Hahaha no worries it was great way..not sure why I missed it..Just fixed that. Thanks @gauravano

gauravano commented on app/views/maps/view_map.html.erb:
```diff
@@ -0,0 +1,94 @@
+              <br>
+              <h3>Projects/ Requests for observations</h3>
+              <br>
+              <p> Community observations and moitoring are a key source of environmental data. Help add to one of the [projects] here. Read more or post your oen call for input</p>
```
```suggestion
              <p> Community observations and monitoring are a key source of environmental data. Help add to one of the [projects] here. Read more or post your oen call for input</p>
```

gauravano commented on app/views/maps/view_map.html.erb:
```diff
@@ -0,0 +1,94 @@
+              <br>
+              <h3>Projects/ Requests for observations</h3>
+              <br>
+              <p> Community observations and moitoring are a key source of environmental data. Help add to one of the [projects] here. Read more or post your oen call for input</p>
```
```suggestion
              <p> Community observations and monitoring are a key source of environmental data. Help add to one of the [projects](<link here>) here. Read more or post your open call for input</p>
```

I think, there's a link to be inserted here.

cesswairimu replied:
Nice catch..thanks...fixed it

gauravano commented:
How about symmetrically aligning the icons and buttons. Also, how about having the same style for both the buttons?

![Screenshot from 2019-07-28 23-46-53](https://user-images.githubusercontent.com/20878070/62011151-3ab82900-b192-11e9-89ca-e05247965e37.png)

Everything else, looks great :100:

cesswairimu commented:
@jywarren I was following the design here #761. Thats why I have different buttons

cesswairimu commented:
![Screenshot from 2019-07-28 22-24-39](https://user-images.githubusercontent.com/17081074/62011849-ce83f800-b186-11e9-96d3-326d17bc974a.png)

@gauravano I aligned the two sections :point_up:

jywarren commented on app/views/maps/view_map.html.erb:
```diff
@@ -0,0 +1,104 @@
+      </div>
+      <div class="row text-center lead-text" id="info">
+          <div class="col-md-3 col-sm-5 offset-2">
+              <p> <%= link_to "Ask a question", 'https://publiclab.org/questions/new', target: 'blank', class: "btn btn-lg btn-primary" %></p>
```
Can we do a URL like: 

```https://publiclab.org/questions/new?tags=lat:<%= @map.lat %>,lon:<%= @map.lon %>&body=Question posted from map at https://mapknitter.org/m/<%= @map.slug %>```

jywarren replied:
and for the "post an observation" as well, below?

cesswairimu replied:
Yeah, this looks great..will try it. Thanks Jeff

jywarren commented on app/views/maps/view_map.html.erb:
```diff
@@ -0,0 +1,104 @@
+              <br>
+              <h3>Ask a question</h3>
+          </div>
+          <div class="col-md-3 col-sm-5 offset-2">
```
I'm a little worried that the titles in one row and the texts in the next row may not fold properly when the window width is reduced -- can you check this in a mobile-width window to be sure? Thank you!

cesswairimu replied:
oh yeah, this will look horrible in a mobile-view, will make the whole thing a row and find some text to add for uniformity

jywarren commented on app/views/maps/view_map.html.erb:
```diff
@@ -0,0 +1,104 @@
+      window.mapKnitter = new MapKnitter.Map({
+        latlng:     L.latLng(<%= @map.lat %>, <%= @map.lon %>),
+        zoom:       <%= @map.zoom %>,
+        readOnly:   true,
```
Is this working yet? It had gotten a bit broken in all the recent changes. We may need to coordinate with @sashadev-sky and maybe @ViditChitkara on https://github.com/publiclab/Leaflet.DistortableImage/issues/351 -- ah, i see solution in https://github.com/publiclab/mapknitter/pull/892 we can review! Looking... in a bit!

sashadev-sky replied:
@jywarren @ViditChitkara @cesswairimu I merged the fix this should be good to go now!

jywarren commented on app/views/front_ui/_maps.html.erb:
```diff
@@ -1,13 +1,11 @@
     <div class="card-deck map col-md-6 col-lg-3 col-xs-12 mb-2 ml-2">
-      <div class="card card-without-border">
+        <div class="card bg-transparent card-without-border">
         <a href="/maps/<%= map.slug %>">
```
Shall we change this to the "view only" view?

jywarren commented:
Finally, i'm thinking about how best to place the "Edit" button. We want this to be as easy as possible... i just worry it's not visible enough on the map:

![image](https://user-images.githubusercontent.com/24359/62350424-5bbea800-b50b-11e9-8ff1-b167fb95967d.png)

Maybe we should put it on the right edge just under the map, kind of opposite the "Embed" button? What do folks think? @publiclab/mapknitter-reviewers

sashadev-sky commented:
@jywarren #892 is ready! The only thing I’m not sure if is bug or not is on page reload, the image appears more zoomed in (it appears as if zoomed to ‘fitbounds’ although I haven’t looked into it. 

Not sure if this was happening before bc the last time I pulled ‘main’ was a while ago and now it is not working for me to check. 

Sasha Boginsky
Goizueta Business School
Emory University | BBA Class of 2017
646.241.5756 | sasha.boginsky@emory.edu 

> On Aug 2, 2019, at 2:50 AM, Jeffrey Warren <notifications@github.com> wrote:
> 
> @jywarren commented on this pull request.
> 
> In app/views/maps/view_map.html.erb:
> 
> > +          <div class="col-md-3 col-sm-5 offset-2">
> +              <p> <%= link_to "Ask a question", 'https://publiclab.org/questions/new', target: 'blank', class: "btn btn-lg btn-primary" %></p>
> +          </div>
> +          <div class="col-md-3 col-sm-5 offset-2">
> +              <p> <%= link_to "Post an Observation", 'https://publiclab.org/post', target: 'blank', class: "btn btn-outline-secondary btn-lg" %></p>
> +
> +          </div>
> +      </div>
> +
> +  <script>
> +      var map
> +      (function(){
> +      window.mapKnitter = new MapKnitter.Map({
> +        latlng:     L.latLng(<%= @map.lat %>, <%= @map.lon %>),
> +        zoom:       <%= @map.zoom %>,
> +        readOnly:   true,
> Is this working yet? It had gotten a bit broken in all the recent changes. We may need to coordinate with @sashadev-sky and maybe @ViditChitkara on publiclab/Leaflet.DistortableImage#351 -- ah, i see solution in #892 we can review! Looking... in a bit!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

cesswairimu commented:
Thanks @sashadev-sky

cesswairimu commented:
![screencapture-localhost-3000-m-nairobi-8aeed88e-b85b-4c86-99ca-71f127d9d75a-2019-08-08-14_38_32](https://user-images.githubusercontent.com/17081074/62700377-689b3a00-b9ea-11e9-95cf-610b6f5dac07.png)

jywarren commented:
Hi! It looks like there's an odd white zone to the right side, do you think
we could address that?

This is really shaping up. Shall we add "by" to the author name under the
title?

Also, shall we make a button to 'return' to this view from the edit view
(i.e. /maps/____)?

We can open a new issue for this, but in follow-up, this page should be the
default. So, shall we make this appear at /maps/____ and move the "edit"
page to /map/edit/_____ -- what do you think?

On Thu, Aug 8, 2019 at 5:41 PM Cess <notifications@github.com> wrote:

> [image:
> screencapture-localhost-3000-m-nairobi-8aeed88e-b85b-4c86-99ca-71f127d9d75a-2019-08-08-14_38_32]
> <https://user-images.githubusercontent.com/17081074/62700377-689b3a00-b9ea-11e9-95cf-610b6f5dac07.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/mapknitter/pull/777?email_source=notifications&email_token=AAAF6J6N7MDX2DQM4GWTJCTQDQA33A5CNFSM4H4WTQUKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD33K5II#issuecomment-519483041>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAF6J7G5CEIOZ3J2GDC5JLQDQA33ANCNFSM4H4WTQUA>
> .
>

cesswairimu commented:
Hi Jeff...Sounds good..addressing that. Thanks

cesswairimu commented:
This is the new edit page `/maps/edit/:id`...Added a button to the show page .
![edit-map](https://user-images.githubusercontent.com/17081074/62864898-a31c1400-bd15-11e9-8acf-4b6f26027cb8.png)


This is now the show page `/maps/:id`
![screencapture-localhost-3000-maps-nairobi-8aeed88e-b85b-4c86-99ca-71f127d9d75a-2019-08-12-15_26_36](https://user-images.githubusercontent.com/17081074/62864935-c5159680-bd15-11e9-80c7-2283344fbc9e.png)

gauravano commented:
Looking great :tada: :100:.  I think these 2 sections can be further improved:

![Screenshot from 2019-08-12 20-18-36](https://user-images.githubusercontent.com/20878070/62874292-94962280-bd3e-11e9-859e-6afc6758790f.png)

![Screenshot from 2019-08-12 20-18-05](https://user-images.githubusercontent.com/20878070/62874293-952eb900-bd3e-11e9-8c8d-c17de44ce5c8.png)

@jywarren thoughts?

cesswairimu commented:
Thanks @gauravano will update that

cesswairimu commented:
@gauravano I made the changes. What do you think? Thanks
*Top buttons*
![top-buttons](https://user-images.githubusercontent.com/17081074/63037349-a22ae400-bec7-11e9-8eb9-68dfcba73e5b.png)



*Question and observation section*
![bottom-buttons](https://user-images.githubusercontent.com/17081074/63037348-a22ae400-bec7-11e9-885a-5dd54accdd54.png)

*Edit page*
![view-button](https://user-images.githubusercontent.com/17081074/63037350-a2c37a80-bec7-11e9-923a-a2ad989e5916.png)

jywarren commented:
Button looks good! Perhaps it should say "« Exit editor" ? It's a bit of a
different UI concept for the page, but if we start thinking of the editable
page as the "map editor" then this button serves to close it, you know?

On Wed, Aug 14, 2019 at 12:17 PM Cess <notifications@github.com> wrote:

> @gauravano <https://github.com/gauravano> I made the changes. What do you
> think? Thanks
> *Top buttons*
> [image: top-buttons]
> <https://user-images.githubusercontent.com/17081074/63037349-a22ae400-bec7-11e9-8eb9-68dfcba73e5b.png>
>
> *Question and observation section*
> [image: bottom-buttons]
> <https://user-images.githubusercontent.com/17081074/63037348-a22ae400-bec7-11e9-885a-5dd54accdd54.png>
>
> *Edit page*
> [image: view-button]
> <https://user-images.githubusercontent.com/17081074/63037350-a2c37a80-bec7-11e9-923a-a2ad989e5916.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/mapknitter/pull/777?email_source=notifications&email_token=AAAF6J2AWTHVQXGHLYYEA4DQEQVY7A5CNFSM4H4WTQUKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4JKFVQ#issuecomment-521315030>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAF6J5BZFEEKOE4CTYODXDQEQVY7ANCNFSM4H4WTQUA>
> .
>

cesswairimu commented:
Oh yeah, that sounds great. I have updated that. Thanks

jywarren commented:
Great - and are the routes changed now too, or will we do that in follow-up? Thanks, Cess!

cesswairimu commented:
The routes are changed too @jywarren 
cc

> This is the new edit page `/maps/edit/:id`...Added a button to the show page .
> ![edit-map](https://user-images.githubusercontent.com/17081074/62864898-a31c1400-bd15-11e9-8acf-4b6f26027cb8.png)
> 
> This is now the show page `/maps/:id`
> ![screencapture-localhost-3000-maps-nairobi-8aeed88e-b85b-4c86-99ca-71f127d9d75a-2019-08-12-15_26_36](https://user-images.githubusercontent.com/17081074/62864935-c5159680-bd15-11e9-80c7-2283344fbc9e.png)

jywarren commented:
OK i think this is almost ready to merge. Cess, I'm interested in removing the "requests for observations" for now, because I think we need to build some more features for it. Can we comment that out, and move the "Questions" one up to be side by side with the `Make your own Map` region? 

In the future, I'm thinking we may need to develop a new type of grid, that is "content near coordinates" -- something like the https://publiclab.org/inline-maps power tags, but showing a table of projects near given coordinates. It could be using the syntax `[content:lat:lon]` maybe... we need some more brainstorming. This is a project we may be interested in working with you on, Cess, if you're interested? Since you've worked on both MapKnitter and the PL API. Let's wrap up this PR and then discuss more?

cesswairimu commented:
That's a great idea :heart: I have updated it. Here is the current display @jywarren 

`image1`
![final-view](https://user-images.githubusercontent.com/17081074/63119339-96fbb500-bfa8-11e9-9b07-73ce40af2362.png)

`image2`
![view2](https://user-images.githubusercontent.com/17081074/63119823-a7f8f600-bfa9-11e9-82e8-1867b9d90a06.png)



I have pushed the changes that have ` image 1` let me know which one you prefer. Thanks

jywarren commented:
Great, i like 1. Let's wait until Rails 5 is merged so as not to create yet
another conflict for them, then we can merge this! Thanks, Cess!

On Thu, Aug 15, 2019 at 3:14 PM Cess <notifications@github.com> wrote:

> That's a great idea ❤️ I have updated it. Here is the current display
> @jywarren <https://github.com/jywarren>
>
> image1
> [image: final-view]
> <https://user-images.githubusercontent.com/17081074/63119339-96fbb500-bfa8-11e9-9b07-73ce40af2362.png>
>
> image2
> [image: view2]
> <https://user-images.githubusercontent.com/17081074/63119823-a7f8f600-bfa9-11e9-82e8-1867b9d90a06.png>
>
> I have pushed the changes that have image 1 let me know which one you
> prefer. Thanks
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/mapknitter/pull/777?email_source=notifications&email_token=AAAF6JZ4BEWDRA6NUNIUWKLQEWTFTA5CNFSM4H4WTQUKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4MXGRQ#issuecomment-521761606>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAF6J5RIPNX3CODWS5MK4DQEWTFTANCNFSM4H4WTQUA>
> .
>

cesswairimu commented:
Hey Jeff, maybe we could merge this now that rails 5 is working fine? Thanks

cesswairimu commented:
Hey @jywarren , maybe we could merge this now that rails 5 is working fine? Thanks

jywarren commented:
Hi Cess! Yes, sounds good. Merging now, can you check it out on stable once we do?

cesswairimu commented:
Hey Jeff, it is working fine on stable apart from the routing error that Sasha has fixed in her latest PR. Thanks
